src/main/java/org/jenkinsci/plugins/DependencyTrack/DependencyTrackPublisher.java

```diff
@@ -178,91 +179,90 @@ public void perform(@Nonnull final Run<?, ?> build,
+                this.projectId = new ProjectLookupParser(jsonResponseBody).parse().getProject().getUuid();
+            }
+            logger.log(Messages.Builder_Findings_Processing());
+            final String jsonResponseBody = apiClient.getFindings(this.projectId);
+            final FindingParser parser = new FindingParser(build.getNumber(), jsonResponseBody).parse();
+            final ArrayList<Finding> findings = parser.getFindings();
+            final SeverityDistribution severityDistribution = parser.getSeverityDistribution();
+            final ResultAction projectAction = new ResultAction(build, findings, severityDistribution);
+            build.addAction(projectAction);
+
+            // Get previous results and evaluate to thresholds
+            final Run previousBuild = build.getPreviousBuild();
+            final RiskGate riskGate = new RiskGate(getThresholds());
+            if (previousBuild != null) {
+                final ResultAction previousResults = previousBuild.getAction(ResultAction.class);
+                if (previousResults != null) {
+                    final Result result = riskGate.evaluate(
+                            previousResults.getSeverityDistribution(),
+                            previousResults.getFindings(),
+                            severityDistribution,
+                            findings);
+                    evaluateRiskGates(build, logger, result);
                 } else { // Resolves https://issues.jenkins-ci.org/browse/JENKINS-58387
                     final Result result = riskGate.evaluate(severityDistribution, new ArrayList<>(), severityDistribution, findings);
                     evaluateRiskGates(build, logger, result);
                 }
+            } else { // Resolves https://issues.jenkins-ci.org/browse/JENKINS-58387
+                final Result result = riskGate.evaluate(severityDistribution, new ArrayList<>(), severityDistribution, findings);
+                evaluateRiskGates(build, logger, result);
             }
-        } catch (ApiClientException e) {
-            logger.log(e.getMessage());
-            build.setResult(Result.FAILURE); //todo: make configurable
         }
     }
 
-    private void evaluateRiskGates(final Run<?, ?> build, final ConsoleLogger logger, final Result result) {
+    private void evaluateRiskGates(final Run<?, ?> build, final ConsoleLogger logger, final Result result) throws AbortException {
         if (Result.SUCCESS != result) {
             logger.log(Messages.Builder_Threshold_Exceed());
-            build.setResult(result); // only set the result if the evaluation fails the threshold
+            if (Result.FAILURE == result) {
+                // attempt to halt the build
+                throw new AbortException("Severity distribution failure thresholds exceeded");
+            } else {
+                // allow build to proceed, but mark overall build unstable
+                build.setResult(result);
```

Comment:
The only place where I think it's valid to set the build result is when setting it to UNSTABLE.  This will allow the build to proceed, but it will be marked unstable when complete.